lib/ffmpeg/encoderbase.py

```diff
@@ -63,6 +63,7 @@ def hwformat(self):
   pict          = property(lambda s: s.ifprop("vpict", " -pic_timing_sei 0"))
   roi           = property(lambda s: s.ifprop("roi", ",addroi=0:0:{width}/2:{height}/2:-1/3"))
   strict        = property(lambda s: s.ifprop("strict", " -strict {strict}"))
+  global_quality = property(lambda s: s.ifprop("global_quality", " -global_quality {global_quality}"))
```

Comment:
No.  Please override `qp` property as needed.  See `ffmpeg/vaapi/encoder.py` for example.

Reply:
Oddly, I'm curious how we got trend model for AV1 to work before... b/c ffmpeg-qsv encoder test is coded to use `-q {qp}`.  Did ffmpeg-qsv middleware recently change this to use `global_quality` parameter?

IDK, maybe I modified this test locally when I generated the models before... and just forgot to file an issue/PR to change it.

Reply:
> No. Please override `qp` property as needed. See `ffmpeg/vaapi/encoder.py` for example.

Is it fine to add/modification cqp testing for ffmpeg-vaapi? it seems that ffmpeg-vaapi doesn't include av1 cqp test path (currently, the existing ffmpeg-vaapi's cqp test cmdling is icq test cmdline)

Reply:
Not quite sure I understand your concern.

> (currently, the existing ffmpeg-vaapi's cqp test cmdling is icq test cmdline)

Do you mean **ffmpeg-qsv**?

The **ffmpeg-vaapi** cmdline has `-rc_mode` option to explicitly set it... so the cmdline is not the same.
https://github.com/intel/vaapi-fits/blob/a2553c256913dafc03c1d5d6f840b66a9972e9fe/lib/ffmpeg/vaapi/encoder.py#L50-L53

> Is it fine to add/modification cqp testing for ffmpeg-vaapi? 

Thus, it would not modify the cqp tests for **ffmpeg-vaapi** at all... that is, the ffmpeg-vaapi encoder class already translates the `qp` to `-global-quality {qp}` for `AV1`.  It does not matter what `rcmode` is specified.
https://github.com/intel/vaapi-fits/blob/a2553c256913dafc03c1d5d6f840b66a9972e9fe/lib/ffmpeg/vaapi/encoder.py#L31-L35


For **ffmpeg-qsv**, however, it looks like `qp` is translated to `-q {qp}` for all `rcmode`.  And I don't see an option/property used to explicitly set `rcmode` on the cmdline.  Thus, that would make it the same cmdline.  Thus, if ffmpeg-qsv `AV1` `icq` uses `-global-quality`, then you only need to add that condition in the `qp` property function for ffmpeg-qsv encoder when rcmode == icq.
https://github.com/intel/vaapi-fits/blob/a2553c256913dafc03c1d5d6f840b66a9972e9fe/lib/ffmpeg/qsv/encoder.py#L36-L43

IMO, it's better to encapsulate these intricate details down at the core of the test code.  Not at the test class or user level.

Reply:
> Thus, it would not modify the cqp tests for **ffmpeg-vaapi** at all... that is, the ffmpeg-vaapi encoder class already translates the `qp` to `-global-quality {qp}` for `AV1`. It does not matter what `rcmode` is specified.

ffmpeg-vaapi av1 checked in test cmd for CQP mode line had issue, it is not right. fixed it in this PR

Reply:
> Not quite sure I understand your concern.
> 
> > (currently, the existing ffmpeg-vaapi's cqp test cmdling is icq test cmdline)
> 
> Do you mean **ffmpeg-qsv**?
> 
> The **ffmpeg-vaapi** cmdline has `-rc_mode` option to explicitly set it... so the cmdline is not the same.
> 
> https://github.com/intel/vaapi-fits/blob/a2553c256913dafc03c1d5d6f840b66a9972e9fe/lib/ffmpeg/vaapi/encoder.py#L50-L53
> 
> > Is it fine to add/modification cqp testing for ffmpeg-vaapi?
> 
> Thus, it would not modify the cqp tests for **ffmpeg-vaapi** at all... that is, the ffmpeg-vaapi encoder class already translates the `qp` to `-global-quality {qp}` for `AV1`. It does not matter what `rcmode` is specified.
> 
> https://github.com/intel/vaapi-fits/blob/a2553c256913dafc03c1d5d6f840b66a9972e9fe/lib/ffmpeg/vaapi/encoder.py#L31-L35
> 
> For **ffmpeg-qsv**, however, it looks like `qp` is translated to `-q {qp}` for all `rcmode`. And I don't see an option/property used to explicitly set `rcmode` on the cmdline. Thus, that would make it the same cmdline. Thus, if ffmpeg-qsv `AV1` `icq` uses `-global-quality`, then you only need to add that condition in the `qp` property function for ffmpeg-qsv encoder when rcmode == icq.
> 
> https://github.com/intel/vaapi-fits/blob/a2553c256913dafc03c1d5d6f840b66a9972e9fe/lib/ffmpeg/qsv/encoder.py#L36-L43
> 
> IMO, it's better to encapsulate these intricate details down at the core of the test code. Not at the test class or user level.

Sure, updated the patch,